presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerde.java

```diff
@@ -62,15 +64,17 @@ public SerializedPage serialize(Page page)
         SliceOutput serializationBuffer = new DynamicSliceOutput(toIntExact(page.getSizeInBytes() + Integer.BYTES)); // block length is an int
         writeRawPage(page, serializationBuffer, blockEncodingSerde);
         Slice slice = serializationBuffer.slice();
+
         int uncompressedSize = serializationBuffer.size();
         byte markers = PageCodecMarker.none();
 
         if (compressor.isPresent()) {
-            ByteBuffer compressionBuffer = ByteBuffer.allocate(compressor.get().maxCompressedLength(uncompressedSize));
-            compressor.get().compress(slice.toByteBuffer(), compressionBuffer);
-            compressionBuffer.flip();
-            if ((((double) compressionBuffer.remaining()) / uncompressedSize) <= MINIMUM_COMPRESSION_RATIO) {
-                slice = Slices.wrappedBuffer(compressionBuffer);
+            int maxCompressedSize = compressor.get().maxCompressedLength(uncompressedSize);
+            ensureCompressionBufferCapacity(maxCompressedSize);
```

Comment:
would you use Arrays.ensureCapacity?

Reply:
Hi Masha, I couldn't find this method in Arrays. Did you mean ArrayList.ensureCapacity? We're not using ArrayList here though..

Reply:
You may need to rebase to latest master. The class is `com.facebook.presto.orc.reader.Arrays`.

Reply:
Oh I see. But this class is in presto-orc and PagesSerde is in presto-main and can't import that class. It seems this class could be moved to presto-spi or airlift-equivalent package since other operators might be able to use the methods in them.

Reply:
That's a good point. How would you feel about moving it to presto-arrays?

Reply:
Actually presto-arrays looks a better place.

Reply:
Moved it to presto-arrays. https://github.com/prestodb/presto/pull/13284 
Once it's in,  I'll change the ensureCompressionBufferCapacity() call to use Arrays class.

Reply:
Updated the PR to use Arrays.ensureCapacity()